### PR TITLE
OneClick download/delete log files (#65)

### PR DESCRIPTION
**Pull request by rus-daniel**

### CODE REVIEW

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 1]
```diff
@@ -77,4 +77,70 @@ public function indexAction()
...
+    /**
+     * Delete requested log file
+     */
+    public function deleteAction()
+    {
+
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        if (file_exists($fileName)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: add also ```is_readable```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 2]
```diff
@@ -77,4 +77,70 @@ public function indexAction()
...
+            header('Content-Description: File Transfer');
+            header('Content-Type: application/octet-stream');
+            header('Expires: 0');
+            header('Cache-Control: must-revalidate');
+            header('Pragma: public');
+            header('Content-Disposition: attachment; filename="' . basename($fileName) . '"');
+
+            $file = file($fileName);
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: Just to challenge your code, is that the most optimized way to grab some lines of very large log file (e.g. system.log weight  1 Go). I let you argue.

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 3]
```diff
@@ -77,4 +77,70 @@ public function indexAction()
...
+                for (; $lines > $pointer; $pointer++) {
+                    $text .= $file[$pointer];
+                }
+                echo $text;
+                exit;
+            } else {
+                header('Content-Length: ' . filesize($fileName));
+                readfile($fileName);
+            }
+        } else {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: Avoid ```else``` in this case by moving this before the if with ```if (!file_exists($fileName)) {```.
Objectif reduce the imbrication

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 4]
```diff
@@ -77,4 +77,70 @@ public function indexAction()
...
+
+    /**
+     * Download requested log file
+     */
+    public function downloadAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        if (file_exists($fileName)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: add also ```is_readable```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 5]
```diff
@@ -77,4 +77,70 @@ public function indexAction()
...
+                    $text .= $file[$pointer];
+                }
+                echo $text;
+                exit;
+            } else {
+                header('Content-Length: ' . filesize($fileName));
+                readfile($fileName);
+            }
+        } else {
+            Mage::getSingleton('core/session')->addError(
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: Create private function missingFile to reduce duplicate code (see l141)

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 6]
```diff
@@ -77,4 +77,70 @@ public function indexAction()
...
+                    Mage::helper('oyst_oneclick')->__('Cannot delete "%s" file.', $params['name'])
+                );
+            } else {
+                unlink($fileName);
+                Mage::getSingleton('core/session')->addSuccess(
+                    Mage::helper('oyst_oneclick')->__('Successfully deleted "%s".', $oystLog)
+                );
+            }
+        } else {
+            Mage::getSingleton('core/session')->addError(
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: Same as above.

File: app/code/community/Oyst/OneClick/controllers/PaymentController.php [hunk 7]
```diff
@@ -147,4 +147,24 @@ public function errorAction()
 
         return null;
     }
+
+    public function downloadAction()
+    {
+        $file =  Mage::getBaseDir('var') . DS . 'log' . DS . 'exception.log';
+
+        if (file_exists($file)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/PaymentController.php]: add also ```is_readable```

File: app/code/community/Oyst/OneClick/controllers/PaymentController.php [hunk 8]
```diff
@@ -147,4 +147,24 @@ public function errorAction()
...
+            header('Content-disposition: attachment; filename=' . basename($file));
+            header('Content-type: text/plain');
+            readfile($file);
+        } else {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/PaymentController.php]: empty else? so it's not needed?

File: app/locale/en_US/Oyst_OneClick.csv [hunk 9]
```diff
@@ -126,3 +126,8 @@
 "Success Cancel Order. Payment Id: ""%s"".","Success Cancel Order. Payment Id: ""%s""."
 
 "FreePay has been installed.","FreePay has been installed."
+"File ""%s"" does not exist.","File ""%s"" does not exist."
```
Comment (mehdichaouch) [inline app/locale/en_US/Oyst_OneClick.csv]: add break line above to separate part of translation

File: app/locale/fr_FR/Oyst_OneClick.csv [hunk 10]
```diff
@@ -126,3 +126,8 @@
 "Success Cancel Order. Payment Id: ""%s"".","Success Annulation de la commande. Référence de paiement : ""%s""."
 
 "FreePay has been installed.","FreePay a été correctement installé."
+"File ""%s"" does not exist.","Le fichier ""%s"" n'existe pas."
```
Comment (mehdichaouch) [inline app/locale/fr_FR/Oyst_OneClick.csv]: add break line above to separate part of translation

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 11]
```diff
@@ -0,0 +1,31 @@
+<div>
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: Add usual [file header](https://github.com/oystparis/oyst-1click-magento/blob/master/app/design/frontend/base/default/template/oyst/oneclick/oneclick.phtml#L1).

**Comment by StefBrito** [comment]
You have to delocate the logs from the initiales informations from Oyst-1C :

image

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml (hunk 11 above)
Comment (mehdichaouch) [reply app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: @rus-daniel you forget this?

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 12]
```diff
@@ -77,4 +77,74 @@ public function indexAction()
...
+     * Download requested log file
+     */
+    public function downloadAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->missingFile($fileName, $params['name']);
+
+        if (file_exists($fileName) && is_readable($fileName)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: Some [stuff to fix](https://travis-ci.org/oystparis/oyst-1click-magento/jobs/335081866#L1243)

Exemple : https://magento.stackexchange.com/a/174668/56191

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 13]
```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * This file is part of Oyst_OneClick for Magento.
+ *
+ * @license http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ * @author Oyst <plugin@oyst.com> <@oyst>
+ * @category Oyst
+ * @package Oyst_OneClick
+ * @copyright Copyright (c) 2017 Oyst (http://www.oyst.com)
+ */
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: add break line after this first block please

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+                            <option value="<?php echo $file; ?>"><?php echo $file; ?></option>
+                        <?php endforeach; ?>
+                    </select>
+                </td>
+                <td>
+                    <input type="number" id="oyst-log-file-rows" name="lines" min="1" max="1000">
+                </td>
+                <td>
+                    <div class="oyst-actions">
+                        <button type="button" onclick="getUrl('<?php echo $this->getButtonUrl('download'); ?>')">Download</button>
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: Add translate for Download

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+                        <button type="button" onclick="getUrl('<?php echo $this->getButtonUrl('delete'); ?>')">Delete</button>
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: Add translate for Delete

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+ */
+/**
+ * @var Oyst_OneClick_Block_Adminhtml_Field_LogActions $this
+ */
+?>
+<div>
+    <table>
+        <thead>
+            <tr class="oyst-log-actions-header">
+                <th>Log file</th>
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: Add translations for those lines.

**Comment by StefBrito** [comment]
@rus-daniel @mehdichaouch 
The row functionality is not working when i want to choose in function of the lines i download a blank page.
So it is only functional when i download all the files.
The delete is working well.

File: app/code/community/Oyst/OneClick/etc/system.xml [hunk 17]
```diff
@@ -466,6 +474,26 @@
...
+                    <label>Log Actions</label>
+                    <frontend_type>text</frontend_type>
+                    <sort_order>50</sort_order>
+                    <show_in_default>1</show_in_default>
+                    <show_in_website>1</show_in_website>
+                    <show_in_store>1</show_in_store>
+                    <fields>
+                        <shipping_costs translate="label">
+                            <label>Download/Delete log files</label>
+                            <frontend_model>oyst_oneclick/adminhtml_field_logactions</frontend_model>
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/etc/system.xml]: Change to ```<frontend_model>oyst_oneclick/adminhtml_field_logActions</frontend_model>```

File: app/locale/fr_FR/Oyst_OneClick.csv [hunk 18]
```diff
@@ -126,3 +126,14 @@
...
+"Cannot delete ""%s"" file.","Impossible de supprimer le fichier ""%s""."
+"Successfully deleted ""%s"".","Supprimé avec succès ""%s""."
+"Download/Delete log files","Télécharger / Supprimer les fichiers journaux"
+"Log Actions","Actions de journal"
+"Log file","Fichier journal"
+"Log rows from EOF","Consigner les lignes d'EOF"
+"Actions","Actes"
+"Delete","Effacer"
+"Download","Télécharger"
```
Comment (mehdichaouch) [inline app/locale/fr_FR/Oyst_OneClick.csv]: add breakline

File: app/locale/en_US/Oyst_OneClick.csv [hunk 19]
```diff
@@ -126,3 +126,14 @@
...
+"Cannot delete ""%s"" file.","Cannot delete ""%s"" file."
+"Successfully deleted ""%s"".","Successfully deleted ""%s""."
+"Download/Delete log files","Download/Delete log files"
+"Log Actions","Log Actions"
+"Log file","Log file"
+"Log rows from EOF","Log rows from EOF"
+"Actions","Actions"
+"Delete","Delete"
+"Download","Download"
```
Comment (mehdichaouch) [inline app/locale/en_US/Oyst_OneClick.csv]: Add breakline

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml (hunk 15 above)
Comment (mehdichaouch) [reply app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: update ?

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml (hunk 14 above)
Comment (mehdichaouch) [reply app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: update ?

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml (hunk 16 above)
Comment (mehdichaouch) [reply app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: update ?

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml (hunk 13 above)
Comment (mehdichaouch) [reply app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: update?

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 20]
```diff
@@ -77,4 +81,169 @@ public function indexAction()
...
+     */
+    public function deleteAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->varien_io = new Varien_Io_File();
+        $this->missingFile($fileName, $params['name']);
+
+
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: delete one empty line

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+/**
+ * This file is part of Oyst_OneClick for Magento.
+ *
+ * @license http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ * @author Oyst <plugin@oyst.com> <@oyst>
+ * @category Oyst
+ * @package Oyst_OneClick
+ * @copyright Copyright (c) 2017 Oyst (http://www.oyst.com)
+ */
+/**
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: add break line above this

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 22]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
     {
         $this->_initAction()->renderLayout();
     }
+
+    /**
+     * Download requested log file
+     */
+    public function downloadAction()
+    {
+        $params   = Mage::app()->getRequest()->getParams();
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: remove useless space before =

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 23]
```diff
@@ -14,6 +14,10 @@
  */
 class Oyst_OneClick_Adminhtml_OneClick_ActionsController extends Mage_Adminhtml_Controller_Action
 {
+    private $varien_io;
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: Use camelCase.

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 24]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+     */
+    public function downloadAction()
+    {
+        $params   = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->varien_io = new Varien_Io_File();
+        $this->missingFile($fileName, $params['name']);
+
+        if (isset($params['rows']) && $this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: replace ```is_readable``` by ```Zend_Loader::isReadable()```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 25]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+            } else {
+                $pointer = $lines - $params['rows'];
+                $text = "";
+                for (; $lines > $pointer; $pointer++) {
+                    $text .= $file[$pointer];
+                }
+                $this->setHeaders($fileName);
+                echo $text;
+            }
+        } elseif ($this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: replace ```is_readable``` by ```Zend_Loader::isReadable()```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 26]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+     */
+    public function deleteAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->varien_io = new Varien_Io_File();
+        $this->missingFile($fileName, $params['name']);
+
+        if ($this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: replace ```is_readable``` by ```Zend_Loader::isReadable()```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 27]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+    public function downloadAction()
+    {
+        $params   = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        $this->varien_io = new Varien_Io_File();
+        $this->missingFile($fileName, $params['name']);
+
+        if (isset($params['rows']) && $this->varien_io->fileExists($fileName) && @is_readable($fileName)) {
+            $file  = $this->getTemporaryFile($fileName, $params['rows']);
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: remove useless space before =

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 28]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+            $lines = count($file);
+
+            if ($lines < $params['rows']) {
+                $this->downloadFullFile($fileName);
+            } else {
+                $pointer = $lines - $params['rows'];
+                $text = "";
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: use simple quote

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 29]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+                    Mage::helper('oyst_oneclick')->__('Cannot delete "%s" file.', $params['name'])
+                );
+            } else {
+                $this->varien_io->rm($fileName);
+                Mage::getSingleton('core/session')->addSuccess(
+                    Mage::helper('oyst_oneclick')->__('Successfully deleted "%s".', $oystLog)
+                );
+            }
+        }
+        $this->_redirectReferer();
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: add break line after if

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 30]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+            );
+        }
+    }
+
+    /**
+     * Set download headers.
+     *
+     * @param $file
+     */
+    private function setHeaders($file, $name = null)
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: @rus-daniel I let you do a refactoring of the download method.

You can inspire of:
```
$fileName = Mage::getBaseDir('var') . DS . 'log' . DS . 'Oyst_OneClick.log';

$this->_prepareDownloadResponse(
    'Oyst_OneClick.log',
    array(
        'type' => 'filename',
        'value' => $fileName,
    )
);

$this->getResponse()->sendHeaders();
```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 31]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+        $this->getResponse()->sendHeaders();
+
+    }
+
+    /**
+     * Create temporary file and retrieve it as an array.
+     *
+     * @param $file
+     * @param $rows
+     * @return array|bool
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: add break line

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 32]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+
+    /**
+     * Create temporary file
+     *
+     * @param $fileHandler
+     * @return mixed
+     */
+    private function createTemporaryFile($fileHandler)
+    {
+        $text = "";
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: simple quote

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 33]
```diff
@@ -77,4 +81,168 @@ public function indexAction()
...
+        while (!@feof($fileHandler)) {
+            $text .= @fgets($fileHandler);
+        }
+
+        $this->temporaryFile = @tmpfile();
+        @fwrite($this->temporaryFile, $text);
+        @fseek($this->temporaryFile, 0);
+        $metaData = @stream_get_meta_data($this->temporaryFile);
+        return $metaData['uri'];
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: break line before return

File: app/locale/fr_FR/Oyst_OneClick.csv [hunk 34]
```diff
@@ -126,3 +126,14 @@
...
 "FreePay has been installed.","FreePay a été correctement installé."
+
+"File ""%s"" does not exist.","Le fichier ""%s"" n'existe pas."
+"Cannot delete ""%s"" file.","Impossible de supprimer le fichier ""%s""."
+"Successfully deleted ""%s"".","Supprimé avec succès ""%s""."
+"Download/Delete log files","Télécharger / Supprimer les fichiers journaux"
+"Log Actions","Actions de journal"
+"Log file","Fichier journal"
+"Log rows from EOF","Consigner les lignes d'EOF"
+"Actions","Actes"
```
Comment (mehdichaouch) [inline app/locale/fr_FR/Oyst_OneClick.csv]: Actions

File: app/locale/fr_FR/Oyst_OneClick.csv [hunk 35]
```diff
@@ -126,3 +126,14 @@
...
 
 "FreePay has been installed.","FreePay a été correctement installé."
+
+"File ""%s"" does not exist.","Le fichier ""%s"" n'existe pas."
+"Cannot delete ""%s"" file.","Impossible de supprimer le fichier ""%s""."
+"Successfully deleted ""%s"".","Supprimé avec succès ""%s""."
+"Download/Delete log files","Télécharger / Supprimer les fichiers journaux"
+"Log Actions","Actions de journal"
+"Log file","Fichier journal"
+"Log rows from EOF","Consigner les lignes d'EOF"
```
Comment (mehdichaouch) [inline app/locale/fr_FR/Oyst_OneClick.csv]: Nombre de lignes depuis la fin du fichier

File: app/locale/fr_FR/Oyst_OneClick.csv [hunk 36]
```diff
@@ -126,3 +126,14 @@
 "Success Cancel Order. Payment Id: ""%s"".","Success Annulation de la commande. Référence de paiement : ""%s""."
 
 "FreePay has been installed.","FreePay a été correctement installé."
+
+"File ""%s"" does not exist.","Le fichier ""%s"" n'existe pas."
+"Cannot delete ""%s"" file.","Impossible de supprimer le fichier ""%s""."
+"Successfully deleted ""%s"".","Supprimé avec succès ""%s""."
+"Download/Delete log files","Télécharger / Supprimer les fichiers journaux"
+"Log Actions","Actions de journal"
+"Log file","Fichier journal"
```
Comment (mehdichaouch) [inline app/locale/fr_FR/Oyst_OneClick.csv]: Fichier de log

File: app/locale/fr_FR/Oyst_OneClick.csv [hunk 37]
```diff
@@ -126,3 +126,14 @@
 "Success Cancel Order. Payment Id: ""%s"".","Success Annulation de la commande. Référence de paiement : ""%s""."
 
 "FreePay has been installed.","FreePay a été correctement installé."
+
+"File ""%s"" does not exist.","Le fichier ""%s"" n'existe pas."
+"Cannot delete ""%s"" file.","Impossible de supprimer le fichier ""%s""."
+"Successfully deleted ""%s"".","Supprimé avec succès ""%s""."
+"Download/Delete log files","Télécharger / Supprimer les fichiers journaux"
```
Comment (mehdichaouch) [inline app/locale/fr_FR/Oyst_OneClick.csv]: Télécharger / Supprimer les fichiers de log

**Comment by mehdichaouch** [comment]
@rus-daniel silencing errors is discouraged, son don't use ```@``` ;)

File: app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml [hunk 38]
```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * This file is part of Oyst_OneClick for Magento.
+ *
+ * @license http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ * @author Oyst <plugin@oyst.com> <@oyst>
+ * @category Oyst
+ *
+ * @package Oyst_OneClick
+ * @copyright Copyright (c) 2017 Oyst (http://www.oyst.com)
```
Comment (mehdichaouch) [inline app/design/adminhtml/default/default/template/oyst/oneclick/logactions.phtml]: please keep tow separate bloc
```
/**
 * This file is part of Oyst_OneClick for Magento.
 *
 * @license http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
 * @author Oyst <plugin@oyst.com> <@oyst>
 * @category Oyst
 *
 * @package Oyst_OneClick
 * @copyright Copyright (c) 2017 Oyst (http://www.oyst.com)
 */

/**
 * @var Oyst_OneClick_Block_Adminhtml_Field_LogActions $this
 */
```

like this.

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 39]
```diff
@@ -14,6 +14,32 @@
...
+     * @var Varien_Io_File
+     */
+    private $varienIo;
+
+    /**
+     * Maximum download size
+     *
+     * @var int
+     */
+    private $maxDownloadSize = 12600000;
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: add comment 
```    private $maxDownloadSize = 12600000; // 12Mo```

File: app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php [hunk 40]
```diff
@@ -77,4 +103,132 @@ public function indexAction()
...
+        $this->baseDir = Mage::getBaseDir('var') . DS . 'log' . DS;
+        $filePath = $this->baseDir . $params['name'];
+
+        $this->varienIo = new Varien_Io_File();
+        $this->missingFile($filePath, $params['name']);
+
+        if ($this->varienIo->fileExists($filePath) && Zend_Loader::isReadable($filePath)) {
+            $content = $this->getFileContent($filePath);
+            $tempFileName = $this->createTemporaryFile($params['name'], $content);
+            $file = @file($this->tempFilePath[$tempFileName]);
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php]: remove the @

**Comment by StefBrito** [comment]
@rus-daniel Did you check my comment about the missing of the image OYST

**Comment by mehdichaouch** [comment]
@StefBrito I leave you valid this PR again ;)

**Comment by StefBrito** [comment]
It is working for me approved

**Pull request merged**